File: src/dataset/go/pangram_Exercism_go_drapala/gemini/GeminiFlash_pangram_v4.go
package pangram

func IsPangram(input string) bool {
	var seenLetters uint32
	var uniqueCount int

	const totalAlphabetLetters = 26

	for _, r := range input {
		if r >= 'A' && r <= 'Z' {
			r += ('a' - 'A')
		}

		if r >= 'a' && r <= 'z' {
			bitPos := r - 'a'
			if (seenLetters & (1 << bitPos)) == 0 {
				seenLetters |= (1 << bitPos)
				uniqueCount++

				if uniqueCount == totalAlphabetLetters {
					return true
				}
			}
		}
	}

	return uniqueCount == totalAlphabetLetters
}

File: src/dataset/cpp/pangram_Exercism_thefullarcticfox/gemini/GeminiFlash_pangram_v4.cpp
#include "pangram.h"
#include <string>
#include <bitset>   // For std::bitset
#include <cctype>   // For std::tolower and std::isalpha

namespace pangram {
	bool	is_pangram(const std::string& str) {
		std::bitset<26> seen_letters;
		int unique_count = 0;

		for (char c : str) {
			// std::tolower and std::isalpha expect an int, and the argument must be representable as unsigned char or EOF.
			// Casting to unsigned char before passing helps avoid potential issues with negative char values.
			unsigned char uc = static_cast<unsigned char>(c);
			if (std::isalpha(uc)) {
				char lower_c = std::tolower(uc);
				// Assuming standard English alphabet 'a' through 'z'
				// This avoids reliance on locale settings and directly maps 'a'-'z' to 0-25.
				if (lower_c >= 'a' && lower_c <= 'z') {
					int index = lower_c - 'a';
					if (!seen_letters.test(index)) {
						seen_letters.set(index);
						unique_count++;
						if (unique_count == 26) {
							return true; // All 26 unique letters found, early exit
						}
					}
				}
			}
		}
		return unique_count == 26;
	}
}  // namespace pangram